httpx/dispatch/http2.py

```diff
@@ -108,18 +110,19 @@ def initiate_connection(self) -> None:
     async def send_data(
         self, stream_id: int, data: bytes, timeout: TimeoutConfig = None
     ) -> None:
-        flow_control = self.h2_state.local_flow_control_window(stream_id)
-        chunk_size = min(len(data), flow_control)
-        for idx in range(0, len(data), chunk_size):
-            chunk = data[idx : idx + chunk_size]
-
-            logger.debug(
-                f"send_data stream_id={stream_id} data=Data(<{len(chunk)} bytes>)"
+        while data:
+            flow_control = self.h2_state.local_flow_control_window(stream_id)
+            chunk_size = min(
+                len(data), flow_control, self.h2_state.max_outbound_frame_size
             )
-
-            self.h2_state.send_data(stream_id, chunk)
-            data_to_send = self.h2_state.data_to_send()
-            await self.stream.write(data_to_send, timeout)
+            if chunk_size == 0:
```

Comment:
Just a comprehension note — I’m wondering why this condition indicates that we should wait for the window to be updated? Is it happening when the max outbound limit is 0? Or when we receive no data? Maybe a link to an appropriate section of the HTTP/2 standard could be useful here, as well as a general comment under “while data:” to explain what the window is, why and when it can change and why we even need to handle it.

Reply:
Added some documentation to the code in new commit.